### ISSUE TITLE
SignatureDoesNotMatch error with special characters

### ISSUE THREAD

**Issue by asterikx** [aws/aws-sdk-go-v2#952]
Confirm by changing [ ] to [x] below to ensure that it's a bug:
- [x] I've gone through the [API reference](https://docs.aws.amazon.com/sdk-for-go/v2/api/)
- [x] I've checked [AWS Forums](https://forums.aws.amazon.com) and [StackOverflow](https://stackoverflow.com/questions/tagged/aws-sdk-go) for answers
- [x] I've searched for [previous similar issues](https://github.com/aws/aws-sdk-go-v2/issues) and didn't find any solution
  
**Describe the bug**
I'm generating pre-signed put URLs and hitting `SignatureDoesNotMatch` errors on special characters.

SignatureDoesNotMatch error:
```
Object Key:  1/o-2bWeAMR/some-filename_(1).pdf
Request URL: https://bucket-name.eu-central-1.amazonaws.com/1/o-2bWeAMR/some-filename_%281%29.pdf?...
```

No error:
```
Object Key:  1/o-2bWeAMR/some-filename.pdf
Request URL: https://bucket-name.eu-central-1.amazonaws.com/1/o-2bWeAMR/some-filename.pdf?...
```

So `(` and/or `)` seems to cause the error in this example.

EDIT: Spaces and Umlauts also lead to this error.

**Version of AWS SDK for Go?**
`v0.30.0`

**Version of Go (`go version`)?**
`1.15`

**To Reproduce (observed behavior)**
```go
req, err := c.PresignPutObject(
  ctx,
  &s3.PutObjectInput{
    Bucket: aws.String("bucket-name"),
    Key:    aws.String("1/o-2bWeAMR/some-filename_(1).pdf"),
  },
  s3.WithPresignExpires(in.ExpiresIn),
)
```

**Expected behavior**
Pre-sign client handles special characters correctly (as `aws-sdk-go` `v1` did)

**Comment by asterikx** [comment]
@jasdel can we fix this somehow for the time being? If you can give me some pointers to what needs to be done, I might be able to contribute.

**Comment by jasdel** [comment]
HI @asterikx I'm investigating this issue, but don't have a solid workaround for this issue yet. It looks like logging for presigned SDK computed signature isn't working. Ive created https://github.com/aws/aws-sdk-go-v2/issues/958 to track request signing not working. We'll most likely need to resolve why presign logging isn't working with PresignClient in order to resolve the issue the request signature.

**Comment by jasdel** [comment]
Looks like PR https://github.com/aws/aws-sdk-go-v2/pull/955 will fix the issue with unable to log request signing for presigned URLs. So we can use that to investigate this issue further.

**Comment by jasdel** [comment]
I think we've resolved the issue in #969, and unfortunately there is no good way to workaround this issue other than to pull in the PR. Once this PR is approved and merged it will be included in the next tagged release.
